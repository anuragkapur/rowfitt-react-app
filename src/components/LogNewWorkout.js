import React from 'react';
import {
  Link,
  Route
} from 'react-router-dom';
import LogSingleDistanceWorkout from "./LogSingleDistanceWorkout";

class LogNewWorkout extends React.Component {
  render() {
    return(
      <main role="main" className="container rowfitt-body-main-below-navbar">
        <h1 className="jumbotron-heading">Log a New Workout</h1>
        <div className="row">
          <div className="col">
            <Link className="btn btn-primary btn-lg btn-block" to="/log-new-workout/single-distance-workout"
                  role="button">Single Distance</Link>
            <Link className="btn btn-primary btn-lg btn-block" to="/" role="button">Single Time</Link>
            <Link className="btn btn-primary btn-lg btn-block" to="/" role="button">Fixed Distance Intervals</Link>
            <Link className="btn btn-primary btn-lg btn-block" to="/" role="button">Fixed Time Intervals</Link>
            <Link className="btn btn-primary btn-lg btn-block" to="/" role="button">Variable Distance Intervals</Link>
            <Link className="btn btn-primary btn-lg btn-block" to="/" role="button">Variable Time Intervals</Link>
          </div>
        </div>
        <Route exact path="/log-new-workout/single-distance-workout" component={LogSingleDistanceWorkout}/>
      </main>
    );
  }
}

export default LogNewWorkout;