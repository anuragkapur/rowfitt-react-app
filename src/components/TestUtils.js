import React from 'react';
import {
  BrowserRouter as Router
} from 'react-router-dom';
import {render} from 'react-testing-library';

export function renderWithRouter(component) {
  return {
    ...render(<Router>{component}</Router>)
  }
}