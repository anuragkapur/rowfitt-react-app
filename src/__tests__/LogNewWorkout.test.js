import React from 'react';
import {cleanup} from 'react-testing-library';
import 'jest-dom/extend-expect';
import LogNewWorkout from "../components/LogNewWorkout";
import {renderWithRouter} from "./TestUtils";

afterEach(cleanup);

test("LogNewWorkout page renders buttons to log all workout types", () => {
  const {getByText, container} = renderWithRouter(<LogNewWorkout/>);
  expect(getByText('Single Distance')).toBeInTheDocument();
  expect(getByText('Single Time')).toBeInTheDocument();
  expect(getByText('Fixed Distance Intervals')).toBeInTheDocument();
  expect(getByText('Fixed Time Intervals')).toBeInTheDocument();
  expect(getByText('Variable Distance Intervals')).toBeInTheDocument();
  expect(getByText('Variable Time Intervals')).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
});