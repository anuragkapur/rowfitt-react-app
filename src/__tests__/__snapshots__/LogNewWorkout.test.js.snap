// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogNewWorkout page renders buttons to log all workout types 1`] = `
<main
  role="main"
  class="container rowfitt-body-main-below-navbar"
>
  <h1
    class="jumbotron-heading"
  >
    Log a New Workout
  </h1>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <a
        class="btn btn-primary btn-lg btn-block"
        role="button"
        href="/log-new-workout/upload-image"
      >
        Upload Image
      </a>
    </div>
  </div>
  <hr />
  <div
    class="row"
  >
    <div
      class="col"
    >
      <a
        class="btn btn-primary btn-lg btn-block"
        role="button"
        href="/log-new-workout/single-distance-workout"
      >
        Single Distance
      </a>
      <a
        class="btn btn-primary btn-lg btn-block"
        role="button"
        href="/"
      >
        Single Time
      </a>
      <a
        class="btn btn-primary btn-lg btn-block"
        role="button"
        href="/"
      >
        Fixed Distance Intervals
      </a>
      <a
        class="btn btn-primary btn-lg btn-block"
        role="button"
        href="/"
      >
        Fixed Time Intervals
      </a>
      <a
        class="btn btn-primary btn-lg btn-block"
        role="button"
        href="/"
      >
        Variable Distance Intervals
      </a>
      <a
        class="btn btn-primary btn-lg btn-block"
        role="button"
        href="/"
      >
        Variable Time Intervals
      </a>
    </div>
  </div>
</main>
`;
